### ISSUE TITLE
[BUG] Leftover Files Extremely Glitchy

### ISSUE THREAD

**Issue by aidan-gibson** [alienator88/Pearcleaner#39]
**Describe the bug**

Crashes and gives different sizes when it comes back up. It initially said I had ~120GB of leftover files; crashed, and on next reopen it said 52 before crashing again. Eternal spinning ball and "not responding" appear.

**To Reproduce**
Open App and hit the leftover files button.

**Expected behavior**

No crash.

**Desktop (please complete the following information):**
 - OS: 14.4
 - Pearcleaner App Mode: Regular
 - Pearcleaner Version: 3.3.2

**Screenshots**

<img width="617" alt="image" src="https://github.com/alienator88/Pearcleaner/assets/20917687/7a2767d7-054b-4ff2-875b-59fdd6c40389">

**Comment by aidan-gibson** [comment]
Also, lots of the "leftover files" are actually still in use by apps on my system. ie /Library/Application Support/org.pqrs is for Karabiner, currently on my system (was last modified a week ago, definitely isn't for some old version of karabiner or something)

**Comment by aidan-gibson** [comment]
`/Users/aidangibson/Library/Application Support/cura` for Ultimaker Cura as well

**Comment by aidan-gibson** [comment]
Also wanting to delete plugins for JetBrains apps `/Users/aidangibson/Library/Application Support/JetBrains`

Thanks tho the app is a great concept!

**Comment by aidan-gibson** [comment]
Ah I see the "not 100% accurate" warning now, sorry! 

Still, app becomes almost unusable when I look at leftover files, maybe because it's registering so many (116GB now)

**Comment by alienator88** [comment]
Hey there, so last update I re-wrote A LOT of the logic to make the app more efficient and I found a couple bugs in there just now in regard to the leftover file search. Just pushed out v3.3.3.
Not sure if it will help in your case yet, but I'm unable to reproduce any issues anymore from my side. Granted, I only have around 50 apps installed, and I keep my system pretty clean, so maybe not too many things to find. Your experience might be slightly different.

And regarding the app becoming unstable, I'm thinking that's happening because of the sheer amount of files it tries to load from that 116GB. It has to store it all in memory.
Normally when the leftover search works okay, it shouldn't be able to find that many gigs of files because I have it set to filter out a bunch of un-related things.

Let me know what issues you're still seeing after updating to latest version.

Also as a caveat, that leftover search is more to give the user an overview of possible files leftover from previous apps, and like you noticed not 100% accurate since the app has no idea what apps a user might've had installed in the past. It's a best guess search really in known folders where apps store their files. The user should go through the list and check what they want to remove manually.

**Comment by aidan-gibson** [comment]
Yep, I get the philosophy of the leftover files section now and think you've done a great job! I just initially couldn't see the "not 100% accurate" warning because the app was frozen in that section.

Using 3.3.4, still unresponsive and not responding in Leftover Files 
<img width="898" alt="image" src="https://github.com/alienator88/Pearcleaner/assets/20917687/990610f1-a1df-4f38-817c-8297b8b04ea4">
can sorta click one thing after a while but then becomes completely unresponsive (using 100% cpu in activity monitor). 

What is it storing in memory? My system has 32GB ram and SSD is 940/995GB.

I've also noticed while it's not responding it's using more and more RAM, while I've been typing this it's been in an unresponsive state and RAM usage has increased from 2 -> 7GB and it's still rising at a rate of ~0.05GB/s. I'm going to kill before it consumes any more. Hope this is helpful, let me know if I can debug it any further, this really is a fantastic app and I want it to succeed.

**Comment by alienator88** [comment]
I'm curious if it's the number of files, I don't have a way to test that many as I don't have a lot on my computer. To put it into perspective, for me it takes about 8 seconds to run the leftover search and finds under 100 items, like 160Mb only.

Can you try something for me?
Open up the Console.app and set `com.alienator88.Pearcleaner` as the subsystem in the search bar. Then start recording and reproduce the problem in the leftover files page. Once it starts freezing up, stop the recording in the Console and see if it captures any errors in that log for when it's trying to load the files.
![Screenshot 2024-04-08 at 8 41 21 AM](https://github.com/alienator88/Pearcleaner/assets/6263626/bc7ca38e-1c22-4d23-9638-cc514f11ddd4)

I'm thinking it might not capture any errors in this case if it's a memory issue.
Can you also confirm for me if the issue with the RAM/CPU only starts when you begin to select checkboxes? Is it fine otherwise if nothing is selected? If that's the case, it might give me an idea where to start looking.

**Comment by alienator88** [comment]
I was doing some tests today with list sizes. I created over 10,000 dummy items in the list and I scrolled through the whole thing to load them all into memory, selected them all, etc. My ram still never cracked 100Mb and CPU at 0%.

![Screenshot 2024-04-09 at 10 08 19 AM](https://github.com/alienator88/Pearcleaner/assets/6263626/1e8f6f00-01eb-4b61-a4ed-6683d6114d5a)

I'm really unsure what is happening on your end, I'd need to see if you get any debug errors from Console.app as per my previous update.
